clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -810,33 +889,17 @@ class SyclKernelFieldChecker
     IsInvalid = Diag.Report(FD->getLocation(), diag::err_bad_kernel_param_type)
                 << FieldTy;
   }
+
   void handleStructType(FieldDecl *FD, QualType FieldTy) final {
-    if (SemaRef.getASTContext().getLangOpts().SYCLStdLayoutKernelParams &&
-        !FieldTy->isStandardLayoutType())
-      IsInvalid =
-          Diag.Report(FD->getLocation(), diag::err_sycl_non_std_layout_type)
-          << FieldTy;
-    else {
-      CXXRecordDecl *RD = FieldTy->getAsCXXRecordDecl();
-      if (!RD->hasTrivialCopyConstructor())
-
-        IsInvalid =
-            Diag.Report(FD->getLocation(),
-                        diag::err_sycl_non_trivially_copy_ctor_dtor_type)
-            << 0 << FieldTy;
-      else if (!RD->hasTrivialDestructor())
-        IsInvalid =
-            Diag.Report(FD->getLocation(),
-                        diag::err_sycl_non_trivially_copy_ctor_dtor_type)
-            << 1 << FieldTy;
-    }
+    IsInvalid = !copyableToKernel(FD, FieldTy);
+  }
+
+  void handleArrayType(const CXXBaseSpecifier &, QualType) final {
+    // FIXME
```

Comment:
Add some text here, what do you intend this fixme to do?

Reply:
Deleted.